airbyte/cloud/_sync_results.py

```diff
@@ -0,0 +1,228 @@
+    _latest_status: JobStatusEnum | None = None
+    _connection_response: ConnectionResponse | None = None
+    _cache: CacheBase | None = None
+
+    @property
+    def job_url(self) -> str:
+        """Return the URL of the sync job."""
+        return f"{self.connection.job_history_url}/{self.job_id}"
+
+    def _get_connection_info(self, *, force_refresh: bool = False) -> ConnectionResponse:
+        """Return connection info for the sync job."""
+        if self._connection_response and not force_refresh:
+            return self._connection_response
+
+        self._connection_response = api_util.get_connection(
+            workspace_id=self.workspace.workspace_id,
+            api_root=self.workspace.api_root,
+            api_key=self.workspace.api_key,
+            connection_id=self.connection.connection_id,
+        )
+        return self._connection_response
+
+    def _get_destination_configuration(self, *, force_refresh: bool = False) -> dict[str, Any]:
+        """Return the destination configuration for the sync job."""
+        connection_info: ConnectionResponse = self._get_connection_info(force_refresh=force_refresh)
+        destination_response = api_util.get_destination(
+            destination_id=connection_info.destination_id,
+            api_root=self.workspace.api_root,
+            api_key=self.workspace.api_key,
+        )
+        return destination_response.configuration
+
+    def is_job_complete(self) -> bool:
+        """Check if the sync job is complete."""
+        return self.get_job_status() in FINAL_STATUSES
+
+    def get_job_status(self) -> JobStatusEnum:
+        """Check if the sync job is still running."""
+        if self._latest_status and self._latest_status in FINAL_STATUSES:
+            return self._latest_status
+
+        job_info = api_util.get_job_info(
+            job_id=self.job_id,
+            api_root=self.workspace.api_root,
+            api_key=self.workspace.api_key,
+        )
+        self._latest_status = job_info.status
+
+        return job_info.status
+
+    def raise_failure_status(
+        self,
+        *,
+        refresh_status: bool = False,
+    ) -> None:
+        """Raise an exception if the sync job failed.
+
+        By default, this method will use the latest status available. If you want to refresh the
+        status before checking for failure, set `refresh_status=True`. If the job has failed, this
+        method will raise a `AirbyteConnectionSyncError`.
+
+        Otherwise, do nothing.
+        """
+        latest_status = self._latest_status
+        if refresh_status:
+            latest_status = self.get_job_status()
+
+        if latest_status in FAILED_STATUSES:
+            raise AirbyteConnectionSyncError(
+                workspace=self.workspace,
+                connection_id=self.connection.connection_id,
+                job_id=self.job_id,
+                job_status=self._latest_status,
+            )
+
+    def wait_for_completion(
```

Comment:
Could this result in too many calls to our servers? I am guessing not many people are doing this unless they are using Airbyte APIs. It would be good to add some telemetry that indicates these calls are coming from PyAirbyte (which might already be there).

Reply:
By default it only pings once every 2 seconds, which is probably more patient than the user would be if they implement their own poling logic. We could expand to 3 but I think I prefer not to poll less than once per 5 seconds...

Wdyt, @bindipankhudi ?

Reply:
yep, 2 seconds sounds good to me!